### PR TITLE
Add a Known Issues section

### LINKED ISSUE #153: Using the plugin on elasticsearch-6.2.2, after a couple of hours, then all the search requests got no responses
We checked the url http://localhost:9214/_nodes/hot_threads?threads=10000&ignore_idle_threads=false, 
and we got the thread stack logs below:

```
0.0% (0s out of 500ms) cpu usage by thread 'elasticsearch[node3][search][T#20]'
 10/10 snapshots sharing following 46 elements
   sun.misc.Unsafe.park(Native Method)
   java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
   java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
   java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:870)
   java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1199)
   java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:209)
   java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:285)
   org.elasticsearch.common.util.concurrent.ReleasableLock.acquire(ReleasableLock.java:55)
   org.elasticsearch.common.cache.Cache.lambda$computeIfAbsent$4(Cache.java:371)
   org.elasticsearch.common.cache.Cache$$Lambda$2714/1659733949.accept(Unknown Source)
   org.elasticsearch.common.cache.Cache$CacheSegment.lambda$get$0(Cache.java:224)
   org.elasticsearch.common.cache.Cache$CacheSegment$$Lambda$2758/1471429601.apply(Unknown Source)
   java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:822)
   java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:834)
   java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:2155)
   org.elasticsearch.common.cache.Cache$CacheSegment.get(Cache.java:215)
   org.elasticsearch.common.cache.Cache.get(Cache.java:344)
   org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:370)
   com.o19s.es.ltr.feature.store.index.Caches.cacheLoad(Caches.java:139)
   com.o19s.es.ltr.feature.store.index.Caches.loadFeatureSet(Caches.java:129)
   com.o19s.es.ltr.feature.store.index.CachedFeatureStore.loadSet(CachedFeatureStore.java:51)
   com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:140)
   com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:45)
   org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:97)
   org.elasticsearch.index.query.AbstractQueryBuilder.toFilter(AbstractQueryBuilder.java:119)
   org.elasticsearch.index.query.BoolQueryBuilder.addBooleanClauses(BoolQueryBuilder.java:409)
   org.elasticsearch.index.query.BoolQueryBuilder.doToQuery(BoolQueryBuilder.java:382)
   org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:97)
   org.elasticsearch.index.query.QueryShardContext.lambda$toQuery$2(QueryShardContext.java:304)
   org.elasticsearch.index.query.QueryShardContext$$Lambda$2740/257320679.apply(Unknown Source)
   org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:316)
   org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:303)
   org.elasticsearch.search.SearchService.parseSource(SearchService.java:707)
   org.elasticsearch.search.SearchService.createContext(SearchService.java:558)
   org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:534)
   org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:330)
   org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:316)
   org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:312)
   org.elasticsearch.search.SearchService$3.doRun(SearchService.java:1002)
   org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:672)
   org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
   org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:41)
   org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
   java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
   java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
   java.lang.Thread.run(Thread.java:748)
   
0.0% (0s out of 500ms) cpu usage by thread 'elasticsearch[node3][search][T#27]'
 10/10 snapshots sharing following 46 elements
   sun.misc.Unsafe.park(Native Method)
   java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
   java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1693)
   java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3323)
   java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1729)
   java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
   org.elasticsearch.common.cache.Cache$CacheSegment.remove(Cache.java:290)
   org.elasticsearch.common.cache.Cache.evictEntry(Cache.java:714)
   org.elasticsearch.common.cache.Cache.lambda$computeIfAbsent$4(Cache.java:372)
   org.elasticsearch.common.cache.Cache$$Lambda$2714/1659733949.accept(Unknown Source)
   org.elasticsearch.common.cache.Cache$CacheSegment.lambda$get$0(Cache.java:224)
   org.elasticsearch.common.cache.Cache$CacheSegment$$Lambda$2758/1471429601.apply(Unknown Source)
   java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:822)
   java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:834)
   java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:2155)
   org.elasticsearch.common.cache.Cache$CacheSegment.get(Cache.java:215)
   org.elasticsearch.common.cache.Cache.get(Cache.java:344)
   org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:370)
   com.o19s.es.ltr.feature.store.index.Caches.cacheLoad(Caches.java:139)
   com.o19s.es.ltr.feature.store.index.Caches.loadFeatureSet(Caches.java:129)
   com.o19s.es.ltr.feature.store.index.CachedFeatureStore.loadSet(CachedFeatureStore.java:51)
   com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:140)
   com.o19s.es.ltr.query.StoredLtrQueryBuilder.doToQuery(StoredLtrQueryBuilder.java:45)
   org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:97)
   org.elasticsearch.index.query.AbstractQueryBuilder.toFilter(AbstractQueryBuilder.java:119)
   org.elasticsearch.index.query.BoolQueryBuilder.addBooleanClauses(BoolQueryBuilder.java:409)
   org.elasticsearch.index.query.BoolQueryBuilder.doToQuery(BoolQueryBuilder.java:382)
   org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:97)
   org.elasticsearch.index.query.QueryShardContext.lambda$toQuery$2(QueryShardContext.java:304)
   org.elasticsearch.index.query.QueryShardContext$$Lambda$2740/257320679.apply(Unknown Source)
   org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:316)
   org.elasticsearch.index.query.QueryShardContext.toQuery(QueryShardContext.java:303)
   org.elasticsearch.search.SearchService.parseSource(SearchService.java:707)
   org.elasticsearch.search.SearchService.createContext(SearchService.java:558)
   org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:534)
   org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:330)
   org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:316)
   org.elasticsearch.search.SearchService$2.onResponse(SearchService.java:312)
   org.elasticsearch.search.SearchService$3.doRun(SearchService.java:1002)
   org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:672)
   org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
   org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:41)
   org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
   java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
   java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
   java.lang.Thread.run(Thread.java:748)
```

### PR DESCRIPTION
**Pull request by nomoa**
Inform users about the cache issues and add a workaround.

closes #153

### CODE REVIEW

**Comment by purbon** [comment]
Good job @nomoa, what about actually having this as a different file? we can always reference this in the readme, otherwise the readme can get longer. What do you think?

**Comment by nomoa** [comment]
@purbon done

File: README.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ If you want to just jump in, go straight to the demo. The demo uses [Ranklib](ht
 
 See the full list of [prebuilt versions](http://es-learn-to-rank.labs.o19s.com). If you don't see a version available, see the link below for building or file a request via [issues](https://github.com/o19s/elasticsearch-learning-to-rank/issues).
 
+**IMPORTANT**: please be sure to read the [known issues](KNOWN_ISSUES.md) before installing this plugin.
```
Comment (purbon) [inline README.md]: I would probably not be so strong, and make an own section, something like



```
**Know issues**
As any other peace of software, this plugin is not except of issues. Please read  the [known issues](KNOWN_ISSUES.md) to know more about the current issues that we are aware of. This file might include workarounds to mitigate them when possible.

```
what do you think?

Comment (nomoa) [reply README.md]: sure :)
(even if this bug is particularly serious)

Comment (purbon) [reply README.md]: is serious and important, but I found the statement in the readme too serious 😄 .... i kinda like the way the known_issues section is written in the elasticserach docs, important bugs, communicated nicely.

**Pull request merged**